client/verta/tests/registry/test_model_dependencies.py

```diff
@@ -0,0 +1,194 @@
+    """ Verify that modules used within a function body are extracted
+    as expected, for a function that is not wrapped in verify_io """
+    func: Callable = dependency_testing_model.unwrapped_predict
+    expected_modules = ['verta.runtime', 'click']
+    extracted_modules: List[str] = [
+        f.__name__ for f in md.list_modules_in_function_body(func)
+    ]
+    assert extracted_modules == expected_modules
+
+
+def test_list_modules_in_function_signature_wrapped(dependency_testing_model) -> None:
+    """ Verify that modules used in function arguments are extracted as
+    expected when the function is wrapped in verify_io.
+    """
+    func: Callable = dependency_testing_model.predict
+    expected_modules = [
+        'calendar',
+        'datetime',
+        'numpy',
+        'google.protobuf.message',
+        'pandas.core.frame',
+    ]
+    extracted_modules: List[str] = [
+        f.__name__ for f in
+        md.list_modules_in_function_signature(func)
+    ]
+    assert extracted_modules == expected_modules
+
+
+def test_list_modules_in_function_signature_unwrapped(dependency_testing_model) -> None:
+    """ Verify that modules used in function arguments are extracted as
+    expected with no function wrappers.
+    """
+    func: Callable = dependency_testing_model.unwrapped_predict
+    expected_modules = [
+        'json.encoder',
+        'collections',
+        'sklearn.base',
+        'cloudpickle.cloudpickle_fast',
+        'requests.exceptions',
+    ]
+    extracted_modules: List[str] = [
+        f.__name__ for f in
+        md.list_modules_in_function_signature(func)
+    ]
+    assert extracted_modules == expected_modules
+
+
+def test_list_modules_in_function_return_type_hint_wrapped(dependency_testing_model) -> None:
+    """ Verify that modules used in function return type hints are extracted
+    as expected when the function is wrapped in verify_io.
+    """
+    func: Callable = dependency_testing_model.predict
+    expected_modules = [
+        'calendar',
+        'datetime',
+        'numpy',
+        'google.protobuf.message',
+        'pandas.core.frame'
+    ]
+    extracted_modules: List[str] = [
+        f.__name__ for f in md.list_modules_in_function_signature(func)
+    ]
+    assert extracted_modules == expected_modules
+
+
+def test_list_modules_in_function_return_type_hint_unwrapped(dependency_testing_model) -> None:
+    """ Verify that modules used in function return type hints are extracted
+    as expected with no function wrappers.
+    """
+    func: Callable = dependency_testing_model.unwrapped_predict
+    expected_modules = [
+        'json.encoder',
+        'collections',
+        'sklearn.base',
+        'cloudpickle.cloudpickle_fast',
+        'requests.exceptions'
+    ]
+    extracted_modules: List[str] = [
+        f.__name__ for f in md.list_modules_in_function_signature(func)
+    ]
+    assert extracted_modules == expected_modules
+
```

Comment:
Aren't these exactly the same as `test_list_modules_in_function_signature_wrapped` and `test_list_modules_in_function_signature_unwrapped`, now?

Reply:
Yeah, these are duplicates now that we dropped the `in` assertions.  If we want to test the return type hint specifically, we'll have to separate out that logic into another function.  If we revisit this to deal with nested type hints, we should do that.  For now, I'll drop these tests.

Reply:
Done.